appendices/extensions.txt

```diff
@@ -58,528 +58,26 @@ not be equivalent to `KHX`, but it is likely to indicate a lesser degree of
-endif::VK_NN_vi_surface[]
-
-
-// == NVIDIA `NV` Vendor Extensions
-//
-// :leveloffset: 2
-
-ifdef::VK_NV_clip_space_w_scaling[]
-include::VK_NV_clip_space_w_scaling.txt[]
-endif::VK_NV_clip_space_w_scaling[]
-
-ifdef::VK_NV_dedicated_allocation[]
-include::VK_NV_dedicated_allocation.txt[]
-endif::VK_NV_dedicated_allocation[]
-
-ifdef::VK_NV_external_memory_capabilities[]
-include::VK_NV_external_memory_capabilities.txt[]
-endif::VK_NV_external_memory_capabilities[]
-
-ifdef::VK_NV_external_memory[]
-include::VK_NV_external_memory.txt[]
-endif::VK_NV_external_memory[]
-
-ifdef::VK_NV_external_memory_win32[]
-include::VK_NV_external_memory_win32.txt[]
-endif::VK_NV_external_memory_win32[]
-
-ifdef::VK_NV_fill_rectangle[]
-include::VK_NV_fill_rectangle.txt[]
-endif::VK_NV_fill_rectangle[]
-
-ifdef::VK_NV_fragment_coverage_to_color[]
-include::VK_NV_fragment_coverage_to_color.txt[]
-endif::VK_NV_fragment_coverage_to_color[]
-
-ifdef::VK_NV_framebuffer_mixed_samples[]
-include::VK_NV_framebuffer_mixed_samples.txt[]
-endif::VK_NV_framebuffer_mixed_samples[]
-
-ifdef::VK_NV_geometry_shader_passthrough[]
-include::VK_NV_geometry_shader_passthrough.txt[]
-endif::VK_NV_geometry_shader_passthrough[]
-
-ifdef::VK_NV_glsl_shader[]
-include::VK_NV_glsl_shader.txt[]
-endif::VK_NV_glsl_shader[]
-
-ifdef::VK_NV_sample_mask_override_coverage[]
-include::VK_NV_sample_mask_override_coverage.txt[]
-endif::VK_NV_sample_mask_override_coverage[]
-
-ifdef::VK_NV_shader_subgroup_partitioned[]
-include::VK_NV_shader_subgroup_partitioned.txt[]
-endif::VK_NV_shader_subgroup_partitioned[]
-
-ifdef::VK_NV_viewport_array2[]
-include::VK_NV_viewport_array2.txt[]
-endif::VK_NV_viewport_array2[]
-
-ifdef::VK_NV_viewport_swizzle[]
-include::VK_NV_viewport_swizzle.txt[]
-endif::VK_NV_viewport_swizzle[]
-
-ifdef::VK_NV_win32_keyed_mutex[]
-include::VK_NV_win32_keyed_mutex.txt[]
-endif::VK_NV_win32_keyed_mutex[]
-
-// :leveloffset: 1
-
-
-// == NVIDIA `NVX` Vendor Extensions
-//
-// :leveloffset: 2
-
-ifdef::VK_NVX_device_generated_commands[]
-include::VK_NVX_device_generated_commands.txt[]
-endif::VK_NVX_device_generated_commands[]
-
-ifdef::VK_NVX_multiview_per_view_attributes[]
-include::VK_NVX_multiview_per_view_attributes.txt[]
-endif::VK_NVX_multiview_per_view_attributes[]
-
-// :leveloffset: 1
+include::meta/obsolete_extensions_guard_macro.txt[]
+
+ifndef::HAS_OBSOLETE_EXTENSIONS[]
+[[extension-appendices-list]]
+== List of Extensions
+endif::HAS_OBSOLETE_EXTENSIONS[]
+ifdef::HAS_OBSOLETE_EXTENSIONS[]
+[[extension-appendices-list]]
+== List of Relevant Extensions
+endif::HAS_OBSOLETE_EXTENSIONS[]
+
+include::meta/relevant_extension_appendices_toc.txt[]
+<<<
+include::meta/relevant_extension_appendices.txt[]
+
+
+ifdef::HAS_OBSOLETE_EXTENSIONS[]
+[[succeeded-extension-appendices-list]]
+== List of Promoted and Obsolete Extensions
+include::meta/obsolete_extension_appendices_toc.txt[]
+<<<
+include::meta/obsolete_extension_appendices.txt[]
+endif::HAS_OBSOLETE_EXTENSIONS[]
```

Comment:
That's problematic, because I have no way of knowing at the time of the script running, which extensions are included and which vk version is being compiled. HAS_OBSOLETE_EXTENSIONS variable is computed at `asciidoctor` run time to do the magic.

Reply:
PS: I can hide the magic and the titles into the metafiles though.

Does it make sense to list deprecated and promoted extensions separately? When I decided to list them together, I thought listing them alphabetically together is more useful to the reader than two lists.